virtual_rainforest/core/config.py

```diff
@@ -0,0 +1,362 @@
+                if len(toml_files) != 0:
+                    files.extend(toml_files)
+                else:
+                    empty_fold.append(path)
+            else:
+                files.append(Path(path))
+
+        else:
+            # Add missing path to list of missing paths
+            not_found.append(path)
+
+    # Check for items that are not found
+    if len(not_found) != 0:
+        log_and_raise(
+            f"The following (user provided) config paths do not exist:\n{not_found}",
+            OSError,
+        )
+    # And for empty folders
+    elif len(empty_fold) != 0:
+        log_and_raise(
+            f"The following (user provided) config folders do not contain any toml "
+            f"files:\n{empty_fold}",
+            OSError,
+        )
+    # Finally check that no files are pointed to twice
+    elif len(files) != len(set(files)):
+        log_and_raise(
+            f"A total of {len(files) - len(set(files))} config files are specified more"
+            f" than once (possibly indirectly)",
+            RuntimeError,
+        )
+
+    return files
+
+
+def load_in_config_files(files: list[Path]) -> dict:
+    """Load in a set of toml files checking that no tags are repeated.
+
+    This function also ensure that no tags are repeated across different toml files.
+
+    Args:
+        files: List of files to be read in and checked for overlapping tags
+    """
+
+    # Preallocate container for file names and corresponding dictionaries
+    file_data: list[tuple[Path, dict]] = []
+    conflicts = []
+
+    # Load all toml files that we want to read from
+    for file in files:
+        # If not then read in the file data
+        with open(file, "rb") as f:
+            try:
+                toml_dict = tomllib.load(f)
+                # Check for repeated entries across previous nested dictionaries
+                for item in file_data:
+                    repeats = check_dict_leaves(item[1], toml_dict, [])
+                    for elem in repeats:
+                        conflicts.append((elem, file, item[0]))
+
+                file_data.append((file, toml_dict))
+            except tomllib.TOMLDecodeError as err:
+                log_and_raise(
+                    f"Configuration file {file} is incorrectly formatted. Failed with "
+                    f"the following message:\n{err}",
+                    RuntimeError,
+                )
+
+    # Check if any tags are repeated across files
+    if len(conflicts) != 0:
+        # If so generate full list of errors
+        msg = "The following tags are defined in multiple config files:\n"
+        for conf in conflicts:
+            msg += f"{conf[0]} defined in both {conf[1]} and {conf[2]}\n"
+        msg = msg[:-1]
+        log_and_raise(msg, RuntimeError)
+
+    # Merge everything into a single dictionary
+    config_dict: dict = {}
+    for item in file_data:
+        dpath.util.merge(config_dict, item[1])
```

Comment:
And here is where we use ChainMap:

```suggestion
    from collections import ChainMap  # This imported at the top
    config_dict = ChainMap(*file_data.values())
```